code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/PresentableMessageMapper.java

```diff
@@ -334,68 +334,25 @@ public void setAutoTrack(final boolean useAutoTrack) {
         }
 
         /**
-         * Dispatches an event to be recorded in Event History.
+         * Dispatches an event to be recorded in Event History. Does not write to event history if
+         * the activity ID cannot be retrieved from proposition info.
          *
          * @param interaction {@code String} if provided, adds a custom interaction to the hash
          * @param eventType {@link MessagingEdgeEventType} to be recorded
          */
         void recordEventHistory(final String interaction, final MessagingEdgeEventType eventType) {
-            if (eventType == null) {
-                Log.debug(
-                        MessagingConstants.LOG_TAG,
-                        SELF_TAG,
-                        "Unable to write event history event, MessagingEdgeEventType was null for"
-                                + " message (%s).",
-                        id);
-                return;
-            }
-            if (propositionInfo == null) {
-                Log.debug(
-                        MessagingConstants.LOG_TAG,
-                        SELF_TAG,
-                        "Unable to write event history event (%s), PropositionInfo is not found for"
-                                + " message (%s)",
-                        eventType.getPropositionEventType(),
-                        id);
-                return;
-            }
-            if (messagingExtension == null) {
+            if (propositionInfo == null || StringUtils.isNullOrEmpty(propositionInfo.activityId)) {
                 Log.debug(
                         MessagingConstants.LOG_TAG,
                         SELF_TAG,
-                        "Unable to send a proposition interaction (%s), MessagingExtension is not"
-                                + " found for message (%s)",
-                        eventType.getPropositionEventType(),
+                        "Unable to write event history event %s, proposition info is "
+                                + "not available for message %s",
+                        eventType.toString(),
```

Comment:
Since null check for `eventType` is not yet done here, this might lead to NPE. We can just use `id` in this case for logging